test/e2e/performance_test.go

```diff
@@ -325,3 +326,78 @@ func withPerfTestSetup(fn func(data *TestData), b *testing.B) {
+		t.Fatalf("Error when creating the perftest client Pod: %v", err)
+	}
+	if err := data.podWaitForRunning(defaultTimeout, "perftest-a", testNamespace); err != nil {
+		t.Fatalf("Error when waiting for the perftest client Pod: %v", err)
+	}
+	if err := data.createPodOnNode("perftest-b", masterNodeName(), perftoolImage, nil, nil, nil, []corev1.ContainerPort{{Protocol: corev1.ProtocolTCP, ContainerPort: qperfPort}}); err != nil {
+		t.Fatalf("Error when creating the perftest server Pod: %v", err)
+	}
+	podBIP, err := data.podWaitForIP(defaultTimeout, "perftest-b", testNamespace)
+	if err != nil {
+		t.Fatalf("Error when getting the perftest server Pod's IP: %v", err)
+	}
+	stdout, _, err := data.runCommandFromPod(testNamespace, "perftest-a", "perftool", []string{"bash", "-c", fmt.Sprintf("qperf %s tcp_bw tcp_lat -t 2", podBIP)})
+	if err != nil {
+		t.Fatalf("Error when running qperf client: %v", err)
+	}
+	stdout = strings.TrimSpace(stdout)
+	t.Logf("Report:\n%s", stdout)
+}
+
+func benchmarkNetworkService(t *testing.T, endpointNode, clientNode string) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svc, err := data.createService("perftest-b", qperfPort, qperfPort, map[string]string{"antrea-e2e": "perftest-b"}, false)
+	if err != nil {
+		t.Fatalf("Error when creating perftest service: %v", err)
+	}
+	if err := data.createPodOnNode("perftest-a", clientNode, perftoolImage, nil, nil, nil, nil); err != nil {
+		t.Fatalf("Error when creating the perftest client Pod: %v", err)
+	}
+	if err := data.podWaitForRunning(defaultTimeout, "perftest-a", testNamespace); err != nil {
+		t.Fatalf("Error when waiting for the perftest client Pod: %v", err)
+	}
+	if err := data.createPodOnNode("perftest-b", endpointNode, perftoolImage, nil, nil, nil, []corev1.ContainerPort{{Protocol: corev1.ProtocolTCP, ContainerPort: qperfPort}}); err != nil {
+		t.Fatalf("Error when creating the perftest server Pod: %v", err)
+	}
+	if err := data.podWaitForRunning(defaultTimeout, "perftest-b", testNamespace); err != nil {
+		t.Fatalf("Error when getting the perftest server Pod's IP: %v", err)
+	}
+	stdout, stderr, err := data.runCommandFromPod(testNamespace, "perftest-a", perftoolContainerName, []string{"bash", "-c", fmt.Sprintf("qperf %s tcp_bw tcp_lat -t 2", svc.Spec.ClusterIP)})
+	if err != nil {
+		t.Fatalf("Error when running qperf client: %v, stderr: %s", err, stderr)
+	}
+	stdout = strings.TrimSpace(stdout)
+	t.Logf("Report:\n%s", stdout)
+}
+
+// TestBenchmarkNetworkServiceLocalAccess measures the bandwidth and latency performance of service
+// traffic between a Pod and an Endpoint on same Node.
+func TestBenchmarkNetworkServiceLocalAccess(t *testing.T) {
+	skipIfNotBenchmarkTest(t)
+	benchmarkNetworkService(t, masterNodeName(), masterNodeName())
+}
+
+// TestBenchmarkNetworkServiceRemoteAccess measures the bandwidth and latency performance of service
```

Comment:
I meant that we should explicitly state per-hop latency and not just latency, as it may be misunderstood as e2e latency, which is kind of common.

Reply:
What do you mean by "per-hop"? For me, "per-hop" means for each IP hop and there may be more than one. Do you mean "one-way" as it latency between the client and the server (as opposed to the round trip time)? I believe this is what `qperf` does and I agree it may be worth mentioning in the comment that it is "one-way" latency.

Reply:
@antoninbas I did not use qperf much and relied on google to know its features. [Found this link ](https://www.opsdash.com/blog/network-performance-linux.html), which incorrectly says that latency is average per-hop latency. My bad.. I did not look into man page or source code. Just checked the source code--looks like it is one-way latency that is reported for both client and server-side. Thank you for the clarification.